### PR TITLE
fix(messaging): load default runtime plan for extra placeholders

### LINKED ISSUE #10967: e2e(messaging): accepted provider extras breadcrumb is not emitted
## Investigation Summary

- [Messaging providers](https://github.com/NVIDIA/NemoClaw/actions/runs/33730208977/job/100568882853) fails its accepted-extras assertion on `3076188279c0ea657ec77aeeabebf1f82061eaaf`.
- The same assertion repeats in the [later main target](https://github.com/NVIDIA/NemoClaw/actions/runs/33761515943/job/100678904598).
- The provider input is accepted far enough to continue, but the in-container parser does not emit the contract breadcrumb proving extra keys reached it.
- This is separate from the Slack `fast-uri` packaging failure in the same job.

## Description

The messaging provider E2E cannot prove that accepted extra fields survive transport into the in-container parser.

Expected: an accepted-extras input produces the documented, machine-checkable breadcrumb.

Actual: the target evaluates the breadcrumb predicate as false.

## Reproduction Steps

1. Run the Messaging providers Docker E2E against either candidate.
2. Exercise the accepted provider-extras fixture.
3. Observe the missing breadcrumb assertion.

## Environment

- GitHub Actions Docker E2E
- Runs: [33730208977](https://github.com/NVIDIA/NemoClaw/actions/runs/33730208977), [33761515943](https://github.com/NVIDIA/NemoClaw/actions/runs/33761515943)

## Debug Output

```text
accepted-extras breadcrumb proves extra keys reached in-container parser: expected false to be true
```

## Acceptance Criteria

- Define one stable parser-level signal for accepted extras.
- Product code and the E2E assertion use the same contract.
- The extras case and rejection cases both pass.

## Candidate Fix

- [PR #11011 — fix(messaging): load default runtime plan for extra placeholders](https://github.com/NVIDIA/NemoClaw/pull/11011), authored by @apurvvkumaria.
- Status at linkage: open and ready for review.
- The exact-candidate [run 33915961523](https://github.com/NVIDIA/NemoClaw/actions/runs/33915961523) still logs all three extra Telegram keys as ignored because the empty runtime-plan path did not fall back to the baked plan.
- #11011 owns the product/parser fix and retains both accepted and rejected key coverage. PR #10996 must not replace this proof with an environment-only assertion.

### LINKED ISSUE #10968: e2e(slack): proof workspace omits hoisted fast-uri
## Investigation Summary

- [Messaging providers](https://github.com/NVIDIA/NemoClaw/actions/runs/33730208977/job/100568882853) fails its installed Slack runtime proof with a missing fast-uri module.
- The same failure repeats on [later main](https://github.com/NVIDIA/NemoClaw/actions/runs/33761515943/job/100678904598) and exact candidate 0ddc3ea52 in [run 33915961523](https://github.com/NVIDIA/NemoClaw/actions/runs/33915961523).
- The installed OpenClaw lock already contains fast-uri. The E2E proof resolves the global OpenClaw symlink without following it, then builds its temporary workspace from the symlink parent instead of the real install root.
- Hoisted production dependencies therefore exist in the installed tree but are absent from the proof workspace.

## Description

The external Slack proof workspace does not mirror the installed OpenClaw dependency layout. Slack startup inside that proof fails before provider behavior can be validated.

Expected: the proof workspace resolves the real installed OpenClaw root and exposes its hoisted production dependencies.

Actual: the proof links only the subtree below the global symlink and cannot resolve fast-uri.

## Reproduction Steps

1. Run the Slack case in the Messaging providers Docker E2E.
2. Start the external installed-runtime proof workspace.
3. Observe Node fail while loading the AJV URI runtime.

## Environment

- GitHub Actions Docker E2E
- Runs: [33730208977](https://github.com/NVIDIA/NemoClaw/actions/runs/33730208977), [33761515943](https://github.com/NVIDIA/NemoClaw/actions/runs/33761515943), and [33915961523](https://github.com/NVIDIA/NemoClaw/actions/runs/33915961523)

## Debug Output

Cannot find module fast-uri from the external Slack proof workspace.

## Acceptance Criteria

- Resolve the real installed OpenClaw package root before constructing the proof workspace.
- Add an installed-tree test that includes a hoisted dependency outside the OpenClaw package subtree.
- Do not add another package or remediation mechanism when the installed lock already contains the dependency.
- Messaging providers passes without workspace-hoisting assumptions.

## Candidate Fix

- [PR #11000 — test(e2e): resolve Slack proof workspace through the real OpenClaw root](https://github.com/NVIDIA/NemoClaw/pull/11000), authored by @AzeelSajjad.
- Status at linkage: open draft.
- #11000 follows the real install root and adds the installed-tree regression test.
- PR #10996 should drop its extra fast-uri packaging and remediation path.

### PR DESCRIPTION
**Pull request by apurvvkumaria**
## Outcome

Startup now loads messaging credential bindings from the baked runtime plan when `NEMOCLAW_MESSAGING_RUNTIME_PLAN_PATH` is empty. Valid extra placeholder keys emit the existing accepted-extras signal, while unrelated keys remain rejected. The live messaging proofs now resolve Slack, Telegram, and WeChat through their installed package roots.

## Reason

The startup wrapper always passes `NEMOCLAW_MESSAGING_RUNTIME_PLAN_PATH` to the parser. An unset host value becomes an empty string, which prevented the Python parser from selecting its default artifact path. The live proofs also imported provider runtimes through symlinks under `--preserve-symlinks`, hiding hoisted dependencies.

### Related issues

Fixes #10967
Fixes #10968

## Changes

- Treat an empty runtime-plan path as absent and use the existing default artifact path.
- Exercise the artifact fallback and accepted-extra placeholder behavior in the existing provider-placeholder tests.
- Resolve installed Slack, Telegram, and WeChat runtimes through their managed package layouts.
- Run the Slack proof with normal Node.js symlink resolution so nested OpenClaw dependencies can use the installed hoisted packages.
- Cover the installed provider layouts and symlink behavior with regression tests.
- Lower the legacy test-file size budget after the focused test became shorter.

## Verification

- `npx vitest run --project e2e-support test/e2e/support/messaging-providers-runtime-proofs.test.ts test/e2e/support/messaging-providers-installed-runtime-paths.test.ts` — 50 tests passed.
- `npm run e2e:assertions:check` — passed with 1,882 direct assertions across 86 files.
- `npm run validate:pr` — passed, including repository checks, formatting, lint, secret scanning, commitlint, and CLI type-checking.
- GitHub commit verification — commit `22bad3d704a95159235b724d2bfd6118c192dbde` is Verified.

## Review notes

This change reads placeholder key names across the startup trust boundary. It does not transfer messaging credential values. Existing format and canonical-prefix checks continue to reject malformed or unrelated names, and the full provider-placeholder test suite preserves those denial cases.

The live messaging E2E test remains the external verification for the accepted-extras signal and installed provider runtime proofs.

---
Signed-off-by: Apurv Kumaria <akumaria@nvidia.com>

### CODE REVIEW

File: test/agents/openclaw/runtime/nemoclaw-start.test.ts [hunk 1]
```diff
@@ -2328,19 +2328,21 @@ describe("provider placeholder refresh (#4251)", () => {
...
     fs.mkdirSync(openclawDir, { recursive: true });
     fs.writeFileSync(configPath, `${JSON.stringify(config, null, 2)}\n`);
-    const fn = extractShellFunctionFromSource(
-      src,
-      "refresh_openclaw_provider_placeholders",
-    ).replaceAll("/sandbox/.openclaw", openclawDir);
+    fs.writeFileSync(runtimePlanPath, JSON.stringify(runtimePlan));
+    const fn = extractShellFunctionFromSource(src, "refresh_openclaw_provider_placeholders")
+      .replaceAll("/sandbox/.openclaw", openclawDir)
+      .replaceAll("/usr/local/share/nemoclaw/messaging-runtime-plan.json", runtimePlanPath);
```
[the remark this answers is not in the document]
Comment (apurvvkumaria) [reply test/agents/openclaw/runtime/nemoclaw-start.test.ts:2345]: This unit test replaces only the default path with an isolated file while the path variable remains empty. It verifies the empty-value fallback without writing to the host `/usr/local/share` tree. The existing live `messaging-providers` E2E test owns the packaged-image boundary. It boots the built image and requires the accepted-extras signal with `TELEGRAM_BOT_TOKEN_AGENT_A`; that signal was the failure reported in #10967. Adding another packaging test here would duplicate that existing boundary, so this finding does not require a code change.

**Review by prekshivyas — APPROVED** [review]
Reviewed the final exact head. Normal CI is green, the exact-head messaging E2E passed for installed Slack, Telegram, and WeChat, CodeRabbit's only thread is resolved, and the independent documentation review found no required docs changes.
